09-reproducible-data.Rmd

```diff
@@ -1 +1,83 @@
+
+### Random Number Seeds  
+
+Some tasks in survey analysis require randomness, such as imputation, model training, or creating random samples. By default, the random numbers that R generates will change each time we rerun the code, making it difficult to reproduce the same results. By "setting the seed," we can control the randomness and ensure that the random numbers remain consistent whenever we rerun the code. Others can use the same seed value to reproduce our random numbers and achieve the same results, facilitating reproducibility.
+
+In R, we can use the `set.seed()` function to control the randomness in our code. Set a seed value by providing an integer to the function:
+
+```r
+set.seed(999)
+
+runif(5)
+```
+
+The `runif()` function generates five random numbers from a uniform distribution. Since the seed is set to `999`, running `runif()` multiple times will always produce the same sequence:
+
+```
+[1] 0.38907138 0.58306072 0.09466569 0.85263123 0.78674676
+```
+
+The choice of the seed number is up to the researcher. For example, this could be the date (`20240102`) or time of day (`1056`) when the analysis was first conducted, a phone number (`8675309`), or the first few numbers that come to mind (`369`). As long as the seed is set for a given analysis, the actual number can be up to the researcher to decide.  However, it is important to note that `set.seed()` should be used *before* random number generation but is only necessary once per program to make the entire program reproducible. For example, we could set the seed at the top of a program where libraries are loaded.
+
+### Descriptive Names and Labels
+
+Something else to assist with reproducible research is using descriptive variable names or labeling data. For example, in the ANES data, the variable names in the raw data all start with `V20` and are a string of numbers. To make things easier to reproduce, we opted to change the variable names to be more descriptive of what they contained (e.g., `Age`). This can also be done with the data values themselves.  One way to accomplish this is by creating factors for categorical data, which can ensure that we know that a value of `1` really means `Female`, for example. There are other ways of handling this, such as attaching labels to the data instead of recoding variables to be descriptive (see Chapter \@ref(c11-missing-data)). As with random number seeds, the exact method is up to the researcher, but providing this information can help ensure your research is reproducible.
+
+### Databases
+
+Researchers may consider creating a database for projects with complex or large data structures to manage the data and any changes. Many databases will allow for a history of changes, which can be useful when recoding variables to ensure no inadvertent errors are introduced. Additionally, a database may be more accessible to pass to other researchers if existing relationships between tables and types are complex to map. 
```

Comment:
What do you think of aggregating all the extra resources and adding them in a "Summary" at the end of the chapter?

Reply:
This is a good resource for reproducible environments: https://solutions.posit.co/envs-pkgs/environments/